package.json

```diff
@@ -7,40 +7,49 @@
     "url": "https://github.com/parse-community/parse-server-example"
   },
   "license": "Apache-2.0",
-  "main": "index.js",
+  "main": "dist/index.js",
   "scripts": {
     "coverage": "nyc --reporter=html --reporter=text npm test",
-    "lint": "eslint --cache ./cloud && eslint --cache index.js && eslint --cache ./spec",
-    "lint-fix": "eslint --cache --fix ./cloud && eslint --cache --fix index.js && eslint --cache --fix ./spec",
-    "prettier": "prettier --write '{cloud,spec}/{**/*,*}.js' 'index.js'",
-    "start": "node index.js",
+    "lint": "eslint --cache .",
+    "lint-fix": "eslint --cache --fix .",
+    "prettier": "prettier --write '{cloud,spec}/{**/*,*}.{ts,js,mjs,cjs}' index.ts",
+    "start": "node dist/index.js",
     "pretest": "cross-env MONGODB_VERSION=${MONGODB_VERSION:=7.0.1} MONGODB_TOPOLOGY=${MONGODB_TOPOLOGY:=standalone} mongodb-runner start -t ${MONGODB_TOPOLOGY} --version ${MONGODB_VERSION} -- --port 27017",
-    "test": "TESTING=true jasmine",
+    "test": "TESTING=true tsx node_modules/jasmine/bin/jasmine",
     "posttest": "mongodb-runner stop --all",
-    "watch": "nodemon index.js"
+    "watch": "nodemon index.ts",
+    "build": "tsc",
+    "preview": "npm run build && npm run start"
   },
   "dependencies": {
     "express": "5.1.0",
     "parse": "6.1.1",
     "parse-server": "8.1.0"
   },
-  "type": "module",
   "devDependencies": {
     "@babel/eslint-parser": "7.27.0",
+    "@eslint/js": "^9.0.0",
     "@semantic-release/changelog": "6.0.3",
     "@semantic-release/commit-analyzer": "13.0.1",
     "@semantic-release/git": "10.0.1",
     "@semantic-release/github": "11.0.1",
     "@semantic-release/npm": "12.0.1",
     "@semantic-release/release-notes-generator": "14.0.3",
+    "@types/jasmine": "^5.1.7",
+    "@types/node": "^22.14.1",
+    "@types/parse": "^3.0.9",
     "cross-env": "7.0.3",
     "eslint": "9.24.0",
+    "globals": "^16.0.0",
     "jasmine": "5.6.0",
     "mongodb-runner": "5.8.2",
     "nodemon": "3.1.9",
     "nyc": "17.1.0",
     "prettier": "3.5.3",
-    "semantic-release": "24.2.3"
+    "semantic-release": "24.2.3",
+    "tsx": "^4.19.3",
+    "typescript": "^5.8.3",
```

Comment:
Please remove range ops by running `npm i -DE typescript@5.8.3` and for other newly added dependencies.